Fix installation on Windows runners

norwd opened the pull request:
## Please describe the change you are making

* Add windows runner to the self test workflow
* Modify `determineOS` to detect windows correctly
* Modify `downloadRelease` to install the `gotestfmt` binary in an OS independent manner

## Your code will be released under [the MIT license](LICENSE.md). Are you in the position, and are you willing to release your code under this license?

Yup :)

Fixes #8

norwd committed:
Correctly identify windows os

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Prevent darwin from being reported as Linux

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Remove unix specific logging and determine .zip extension on windows

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Add selftest on windows

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Run self test on all push events (for testing)

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Inline postfix declaration

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Separate extension detection from postfix

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Correctly avoid assignment to const variable

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Make postfix constant

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Prevent curl from assuming tar.gz format

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Refactor download stage

* Separate logical steps
* Use string interpolation for curl command

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Use environment defined temp path

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Fix missing `$`

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Correct all references to hardcoded filename

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Attempt debugging missing download

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Restore original console loggers

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
USe GOOs

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Extract binary directly to GOPATH rather than hardcoded lib path

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Properly log GOOS detection

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Remove GOOS detection as GOOS not in environment

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Replace GOPATH with system dependent lib path

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Fix windows path separator problems

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Simplify archive path

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Fix escape character

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Use more meaningful name for binpath

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Fix whitespace

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Add debugger lines

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Add debugger lines for executed commands

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Use console debug instead of log

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Use program files as binpath

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Add install directory to path

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Remove debuggers

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Fix os check

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Remove redundant pathsep variable

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Add documentation comments

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Fix missing path separators

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Explicitly echo bin path into GITHUB_PATH

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Quote bin path when `cd`ing into it to avoid shell globbing

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Ensure bin path exists

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Attempt to prevent existing directories causing problems

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Use core.addPath to add binpath to job's PATH

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Attempt to use pre-existing lib path instead of creating new one

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Add debugger for PATH variable

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
List System32 files to look for gotestfmt binary

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Remove debugger

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Manually setup binpath and add it to system path

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
List PATH targets for debugging

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Add debuggers for userprofile

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
List install target's contents

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
List files recursively and verbosely

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Search `/bin` specifically

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Only manually add the binpath on windows

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
List all PATHs

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Split on correct separator

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Don't force the directory to exist/into PATH

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Attempt to create custom binpath in windows

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Fix mkdir invocation

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Create all path components

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
List all files in target directory

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Try using pwsh for Windows runner

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Try using exe for pwsh

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Remove testing step

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
List files extracted to bin path

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Expand debugger

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
List files in temporary directory

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
List files in current directory

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Rather than changing directories, specify target to tar command

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Quote parameters to tar command

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Specify target directory before target files

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Move extracted binary manually

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Add callback to file move

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Use system defined target file

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Correct interpolation

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Copy binfile instead of renaming to prevent cross-device linking

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
List installed files

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Use binfile in final success log message

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Remove explicit path separator

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Specify fs.constants.COPYFILE_EXCL copy mode

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Extract archive directly into workspace

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Remove redundant `binfile` variable

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Remove debugger

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
List all files in workspace

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd committed:
Remove debugging steps

Signed-off-by: Yuri Norwood <106889957+norwd@users.noreply.github.com>

norwd commented:
This took waaaaaay too many attempts :joy:

engelmi commented on index.js:
```diff
@@ -9,31 +9,46 @@ async function downloadRelease(octokit, os, org, repo, release, token) {
-            console.log("Linking gotestfmt...")
-            execSync("sudo ln -s /usr/local/lib/gotestfmt/gotestfmt /usr/local/bin/gotestfmt")
+
+            // Extract the archive into the install target
```
Minor: Please remove the inline comments. They are not needed in these cases as the code is self-explanatory. Also, over time comments tend to drift apart from the code leading to confusion.

norwd replied:
Fixed in [ef20a3a](https://github.com/GoTestTools/gotestfmt-action/pull/10/commits/ef20a3ae63b6c43b73c4811e1c7792c43d63a443)

engelmi commented:
> This took waaaaaay too many attempts joy

But the end results looks really good :smiley:

norwd committed:
Remove inline comments

This is per @engelmi's comment: https://github.com/GoTestTools/gotestfmt-action/pull/10/files#r1017614705

Signed-off-by: Y. Meyer-Norwood <106889957+norwd@users.noreply.github.com>
